<launch>

    <!-- Arguments -->
        <arg name="world" default="two"/>

        <!-- Gazebo-->
        <arg name="gazebo_gui" default="true"/>
        <arg name="world_file" default="$(find kobuki_navigation)/worlds/$(arg world).world"/>
        
        <!-- RTAB-->
        <arg name="database_path"     default="rtabmap.db"/>
        <arg name="sw_registered"     default="false"/>
        <arg name="args"  default="--delete_db_on_start"/>
        <arg name="rgb_topic"   default="/camera/rgb/image_raw"/>
        <arg name="depth_topic" default="/camera/depth/image_raw"/>
        <arg name="camera_info_topic" default="/camera/rgb/camera_info"/>
        <arg name="wait_for_transform"  default="0.2"/>


    <!-- Gazebo -->
    <include file="$(find kobuki_navigation)/launch/includes/gazebo.launch">
        <arg name="gui" value="$(arg gazebo_gui)" />
        <arg name="world_file" value="$(arg world_file)" />
    </include>
    

    <!-- SLAM - RTAB -->
    <include file="$(find kobuki_navigation)/launch/includes/rtab.launch">
        <arg name="database_path"       value="$(arg database_path)"/>
        <arg name="sw_registered"       value="$(arg sw_registered)"/>
        <arg name="args"                value="$(arg args)"/>
        <arg name="rgb_topic"           value="$(arg rgb_topic)"/>
        <arg name="depth_topic"         value="$(arg depth_topic)"/>
        <arg name="camera_info_topic"   value="$(arg camera_info_topic)"/>
        <arg name="wait_for_transform"  value="$(arg wait_for_transform)"/>
    </include>


    <!-- RViz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find kobuki_navigation)/rviz/robot.rviz"/>


</launch>